expose/local/local_test.go

```diff
@@ -0,0 +1,71 @@
+package local
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/taskcluster/generic-worker/expose"
+)
+
+func TestConstructor(t *testing.T) {
+	exposer, err := New(net.ParseIP("127.0.0.1"))
+	if err != nil {
+		t.Fatalf("Constructor returned an error: %v", err)
+	}
+	if exposer == nil {
+		t.Fatalf("Constructor did not return a value")
+	}
+}
+
+func makeExposer(t *testing.T) expose.Exposer {
+	exposer, err := New(net.ParseIP("127.0.0.1"))
+	if err != nil {
+		t.Fatalf("Constructor returned an error: %v", err)
+	}
+	return exposer
+}
+
+func makeTestServer() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "Hello, world")
+	}))
+}
+
+func TestExposeHTTP(t *testing.T) {
+	ts := makeTestServer()
+	defer ts.Close()
+	testURL, _ := url.Parse(ts.URL)
+	_, testPortStr, _ := net.SplitHostPort(testURL.Host)
+	testPort, _ := strconv.Atoi(testPortStr)
+
+	exposer := makeExposer(t)
+	exposure, err := exposer.ExposeHTTP(uint16(testPort))
+	if err != nil {
+		t.Fatalf("ExposeHTTP returned an error: %v", err)
+	}
+	defer exposure.Close()
+
+	gotURL := exposure.GetURL()
+	host, port, _ := net.SplitHostPort(gotURL.Host)
+	assert.Equal(t, "http", gotURL.Scheme, "Should return URL with correct schem")
+	assert.Equal(t, "127.0.0.1", host, "Should return URL with correct host")
+	assert.NotEqual(t, "90", port, "Should return URL with a random port")
+
+	res, err := http.Get(gotURL.String())
+	if err != nil {
+		t.Fatal(err)
+	}
+	greeting, err := ioutil.ReadAll(res.Body)
+	res.Body.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, "Hello, world", string(greeting), "got greeting via proxy")
```

Comment:
Let's also assert a 200 status code. Perhaps a non-200 status code test would also be good, both with and without a response body.

Reply:
I'll add that to proxy_test.go.